src/components/landing/useSlideObserver.ts

```diff
@@ -0,0 +1,32 @@
+import { useEffect, useRef } from 'react';
+
+export const useSlideObserver = (
+  setState: (stateNumber: number) => void,
+  stateNumber: number,
+): React.MutableRefObject<HTMLElement | null> => {
+  const isRef = useRef<HTMLElement | null>(null);
+
+  useEffect(() => {
+    let observer: IntersectionObserver;
+    const observePoint: IntersectionObserverCallback = (entries, observer) => {
+      entries.forEach(async (entry) => {
+        if (entry.isIntersecting) {
+          setState(stateNumber);
+          observer.observe(entry.target);
+        }
+      });
+    };
+
+    if (isRef.current) {
+      observer = new IntersectionObserver(observePoint, {
+        threshold: 0.9,
+      });
+      observer.observe(isRef.current);
+    }
+    return () => {
+      observer && observer.disconnect();
+    };
+  }, [setState, stateNumber]);
+
+  return isRef;
+};
```

Comment:
위에 observe 를 상위 스코프로 올리고,

```suggestion
    return () => {
      observer && observer.disconnect();
    };
  }, [setState, stateNumber]);

  const stopObserving1s = () => {
    observer.disconnect();
    setTimeout(()=>{observer.observe(isRef.current);}, 1000);
  };

  return { isRef, stopObserving1s };
};
```

이렇게 observe 껐다 키는 로직 구현해서, 슬라이더 버튼 누를 때 해당 함수 실행시켜도 되겠다!

Reply:
근데 이럴라면

```javascript
  const firstSlideRef = useSlideObserver(setSlideNumber, 1);
  const secondSlideRef = useSlideObserver(setSlideNumber, 2);
  const thirdSlideRef = useSlideObserver(setSlideNumber, 3);
  const fourthSlideRef = useSlideObserver(setSlideNumber, 4);
  const fifthSlideRef = useSlideObserver(setSlideNumber, 5);
  const sixthSlideRef = useSlideObserver(setSlideNumber, 6);
```

이것부터 좀 리팩토링 해야할듯,,!?~

Reply:
단순 1, 2, 3, 4, 5, 6 이니까- 그냥 `useSliderObserver` 내부에서 처리해도 좋을 것 같아!

```suggestion
const { firstSlideRef, secondSlideRef, ..., sixthSlideRef, stopObserving1s } = useSliderObserver(setSlideNumber);
```

이렇게 사용할 수 있게?

Reply:
근데 저라면 ,,,

>개인적으로는 랜딩페이지니까 각 섹션별로의 height 정보의 객체를 stores 폴더에 빼낸 다음에, scrollY 값을 점검하면서 setNumber 를 시키는 게 더 깔끔하지 않을까 싶긴 해요 !!!